Score card fc

RamkrishnaRocket1 opened the pull request:
<!-- Thank you for submitting a PR to Zowe! To help us understand, test, and give feedback on your code, please fill in the details below. -->

## Proposed changes
<!-- Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue. -->

This PR addresses Issue: [*Link to Github issue within https://github.com/zowe/zlux/issues* if any]

This PR depends upon the following PRs:

## Type of change
Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Change in a documentation
- [ ] Refactor the code 
- [ ] Chore, repository cleanup, updates the dependencies.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## PR Checklist
Please delete options that are not relevant.
- [ ] If the changes in this PR are meant for the next release / mainline, this PR targets the "staging" branch.
- [ ] My code follows the style guidelines of this project (see: [Contributing guideline](https://github.com/zowe/zlux/blob/master/CONTRIBUTING.md))
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes
- [ ] video or image is included if visual changes are made
- [ ] Relevant update to CHANGELOG.md
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works, or describe a test method below

## Testing
<!-- Describe how this code should be tested. I've you've added an automated or unit test, then describe how to run it. Otherwise, describe how you have tested it and how others should test it. -->

## Further comments
<!-- If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, or if there are follow-up tasks and TODOs etc... -->

RamkrishnaRocket1 committed:
Create actions

Signed-off-by: RamkrishnaRocket1 <rdakwa@rocketsoftware.com>

RamkrishnaRocket1 committed:
Delete .github/actions

Signed-off-by: RamkrishnaRocket1 <rdakwa@rocketsoftware.com>

RamkrishnaRocket1 committed:
adding actions.yml

Signed-off-by: RamkrishnaRocket1 <rdakwa@rocketsoftware.com>

RamkrishnaRocket1 committed:
folder_strcture_upadte

Signed-off-by: RamkrishnaRocket1 <rdakwa@rocketsoftware.com>

RamkrishnaRocket1 committed:
Update action.yml

Signed-off-by: RamkrishnaRocket1 <rdakwa@rocketsoftware.com>

RamkrishnaRocket1 committed:
Update action.yml

Signed-off-by: RamkrishnaRocket1 <rdakwa@rocketsoftware.com>

RamkrishnaRocket1 committed:
Create ScoreCard-insights.yml

Signed-off-by: RamkrishnaRocket1 <rdakwa@rocketsoftware.com>

RamkrishnaRocket1 committed:
Update ScoreCard-insights.yml

Signed-off-by: RamkrishnaRocket1 <rdakwa@rocketsoftware.com>

struga0258 commented:
I have reviewed this pr and I have some concerns with it. 

1) the schedule has been set to run every 3 hour, I think that is way to frequent. Maybe every week would make more sense.
2) Its also only running on push or schedule, this should run on pull request as well. That way we can see if the pull request breaks anything. for example this pr should have some scans.
3) You have created a repo matrix but only contains like 5-6 repos, why only those? which brings my next concern 4) shouldn't this file or workflow action be added separately into each repo? this would merge all the results into one?  you could still have one action.yaml file here and action.yaml in each repo to trigger this one for the specific repo. 
5) you have created your own ossf-scorecard action but I dont get why? you can just use the main action created by them, simply by doing something like 
uses: ossf/scorecard-action@

RamkrishnaRocket1 committed:
Update action.yml

Signed-off-by: RamkrishnaRocket1 <rdakwa@rocketsoftware.com>

RamkrishnaRocket1 committed:
Delete .github/workflows/ScoreCard-insights.yml

Signed-off-by: RamkrishnaRocket1 <rdakwa@rocketsoftware.com>

RamkrishnaRocket1 commented:
Hi James thanks for reviewing the PR here are the updates 
1. I have updated the workflow to execute on pull request as well to execute as per schedule for once a month -> https://github.com/zowe/zlux-editor/pull/361
2.Actually Yeah its better to call the action for separate repo as per requirement so I have removed the matrix workflow file.
3. So the purpose of creating a separate action file was i needed some custom insights and create a issue post filter as per image 
<img width="1259" height="806" alt="image" src="https://github.com/user-attachments/assets/8a8a0041-2480-4d81-99c0-7174206e777d" />

RamkrishnaRocket1 committed:
Updates in the basic call syntax

Signed-off-by: RamkrishnaRocket1 <rdakwa@rocketsoftware.com>

RamkrishnaRocket1 committed:
Minimum changes

Signed-off-by: RamkrishnaRocket1 <rdakwa@rocketsoftware.com>

RamkrishnaRocket1 committed:
Sarif results 

Signed-off-by: RamkrishnaRocket1 <rdakwa@rocketsoftware.com>

RamkrishnaRocket1 committed:
Details of repo 

Signed-off-by: RamkrishnaRocket1 <rdakwa@rocketsoftware.com>

RamkrishnaRocket1 committed:
Update in checks

Signed-off-by: RamkrishnaRocket1 <rdakwa@rocketsoftware.com>

RamkrishnaRocket1 committed:
Update OSSF Scorecard action to specific commit

Signed-off-by: RamkrishnaRocket1 <rdakwa@rocketsoftware.com>
